File: levels/01/03-OddSquaresOnly/solution_test.go
package fizzbuzz0101

import (
	"fmt"
	"github.com/stretchr/testify/assert"
	"testing"
)

type KeepWriter struct {
	result []interface{}
}

func (k *KeepWriter) Println(a ...interface{}) {
	for _, v := range a {
		k.result = append(k.result, v)
	}
	fmt.Println(a...)
	return
}

func Test_Solution(t *testing.T) {

	fake := new(KeepWriter)
	Solution(fake)
	assert.Equal(t, []interface{}{27,125,343,729,1331,2197,3375,4913,6859,9261,12167,15625}, fake.result)

}

File: levels/01/03-OddSquaresOnly/solution_test.py
from solution import solution
import unittest


def print_gen():
    result = []
    def p(a):
        result.append(a)
        print(a)
    return [p, result]


class TestStringMethods(unittest.TestCase):

    def test_solution(self):
        [fake, result] = print_gen()
        solution(fake)
        expected = [27,125,343,729,1331,2197,3375,4913,6859,9261,12167,15625]
        self.assertEqual(result, expected)


if __name__ == '__main__':
    unittest.main()